clients/src/main/java/org/apache/kafka/common/header/RecordHeader.java

```diff
@@ -14,33 +14,27 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.kafka.common.record;
+package org.apache.kafka.common.header;
 
-import org.apache.kafka.common.utils.Utils;
-
-import java.nio.ByteBuffer;
+import java.util.Arrays;
 import java.util.Objects;
 
-public class Header {
+public class RecordHeader implements Header {
     private final String key;
-    private final ByteBuffer value;
+    private final byte[] value;
```

Comment:
On seeing what we can do to try alleviate your concern as much as possible we can make it so it uses ByteBuffer, again we should note though, that on Header.value() will incure a memory copy still, as simply we move when this copy occurs. As such its only incurred if the header is read.